### ISSUE TITLE
all builtin types are subclassable, even when they shouldn't be

### ISSUE THREAD

**Issue by toshok** [dropbox/pyston#596]
the following example:

```python
class foo(slice):
    pass
```

succeeds in pyston and raises in cpython:

```
TypeError: Error when calling the metaclass bases
    type 'slice' is not an acceptable base type
```

this is due to the fact that Py_TPFLAGS_BASETYPE is or'ed into the tp_flags of all BoxedClasses.  we need a way to either keep that from happening or override it after BoxedClass creation time.  we also need to audit our builtins to make sure only the proper ones are subclassable.

**Comment by undingen** [comment]
while ```slice``` is now fixed there are still some types left:
e.g:
```python
class C(type(xrange(1))): pass
```

**Comment by undingen** [comment]
fixed in #1061
